internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -137,18 +136,37 @@ spec:
             fbcOptIn="false"
             mustSignIndexImage="false"
             mustPublishIndexImage="false"
+        else
+            # check if this fbc fragment is opt-in
+            echo "Fetching the image bundle from $(params.fbcFragment)..."
+            PULL_SPEC_LIST=$(opm render "$(params.fbcFragment)" | jq -r \
+            'select(.schema == "olm.bundle") | "\(.image)" | split("@")[0]' |uniq)
+            for PULL_SPEC in ${PULL_SPEC_LIST}; do
+                # make sure they query is done using the internal name instead of the public
+                PULL_SPEC="${PULL_SPEC//registry.redhat.io/registry.access.redhat.com}"
 
-        # in case of more than one image all should have the same fbc_opt_in value
-        elif [ $(wc -w <<< ${fbcOptIn[@]}) -ne 1 ]; then
-            fbcOptIn="false"
-            mustSignIndexImage="false"
-            mustPublishIndexImage="false"
+                echo "Attempting to fetch from ${FETCH_URL} to check if fragment is \`fbc_opt_in==true\`..."
+                IFS=" " read -r -a fbcOptIn <<< "$(isfbcOptIn "${PULL_SPEC}")"
+            done
+
+            IFS=" " read -r -a fbcOptIn <<< "$(printf "%s\n" "${fbcOptIn[@]}" |sort |uniq)"
```

Comment:
Please take a look in the new [PR](https://github.com/hacbs-release/app-interface-deployments/pull/236) (although it is a draft as it will need rebase) but I made a few changes in this part there, and I also added some comments to explain the logic. 
As Johnny mentioned, I won´t change the logic of the fbcOptIn in this PR. The small code moved around was just to make sure the `fbcOptIn` function would run only when necessary.

Reply:
Anyhow I will take in consideration the comment above in the new PR.

Reply:
OK, but why was this:
```
fbcOptIn=($(printf "%s\n" ${fbcOptIn[@]} |sort |uniq))
```
changed to this?
```
IFS=" " read -r -a fbcOptIn <<< "$(printf "%s\n" "${fbcOptIn[@]}" |sort |uniq)"
```

Reply:
> Please take a look in the new https://github.com/hacbs-release/app-interface-deployments/pull/236

But it still has this weird array saving, instead of just one value. I will comment there.

Reply:
Hum.. this should not be here  anymore. I missed removing it when reverting to the original code. I will follow up there once I revert this piece of code.

Reply:
ok, let me know what it's reverted to the original

Reply:
resolving this one as the PR is approved.